### DISCUSSION TITLE
`.Mismatch` confusion in Node segment

### DISCUSSION THREAD

**Discussion by lucascosti** [JanDeDobbeleer/oh-my-posh#3994] [Troubleshoot] [answered]
Not sure if this is a bug, hence a discussion instead of an issue. 🙂 

The [`.Mismatch` boolean template property in the Node segment](https://ohmyposh.dev/docs/segments/node#properties-1) behaves the opposite of what I expect it would (based on the property name).

If `.Mismatch` returns `true`, I expect that there _**is**_ a mismatch between the Node versions in `.nvmrc` and the environment.

However, according the docs, as well the current behavior, `.Mismatch` returns `true` when "the version in `.nvmrc` matches with `.Full`".

For example, here is what I currently have to do to show the segment only when there is a mismatch of the versions:

```json
{
  "type": "node",
  "style": "powerline",
  "powerline_symbol": "",
  "foreground": "#ffffff",
  "background": "#6CA35E",
  "template": "{{ if not .Mismatch }}{{ .Major }} != {{ .Expected }} {{ end }}"
},
```

**Comment by JanDeDobbeleer** [comment, marked as answer]
@lucascosti fair remark. I think this is because of a refactor. I'll adjust as it's completely counterintuitive.

**Reply by JanDeDobbeleer** [reply]
https://github.com/JanDeDobbeleer/oh-my-posh/pull/3997

**Reply by lucascosti** [reply]
Many thanks, @JanDeDobbeleer! 🙇‍♂️ 🚀
